src/components/app/ProfileSharing.js

```diff
@@ -171,23 +187,40 @@ class ProfileSharingCompositeButton extends PureComponent<
     });
   }
 
-  _attemptToShare() {
-    if (this.state.state !== 'local' && this.state.state !== 'error') {
+  _attemptToShare(isSecondShare: boolean = false) {
+    if (
+      ((!isSecondShare && this.state.state !== 'local') ||
+        (isSecondShare && this.state.state !== 'public')) &&
+      this.state.state !== 'error'
+    ) {
       return;
     }
     this._notifyAnalytics();
 
-    const { profile, predictUrl } = this.props;
+    const { profile, predictUrl, profileSharingStatus } = this.props;
+    const { shareNetworkUrls } = this.state;
 
     new Promise(resolve => {
       if (!profile) {
         throw new Error('profile is null');
       }
-      const jsonString = serializeProfile(profile, this.state.shareNetworkUrls);
+      const urlShareOption = isSecondShare
+        ? !profileSharingStatus.sharedWithUrls
+        : shareNetworkUrls;
+      const jsonString = serializeProfile(profile, urlShareOption);
       if (!jsonString) {
         throw new Error('profile serialization failed');
       }
 
+      const newProfileSharingStatus = {
+        sharedWithUrls: shareNetworkUrls
+          ? true
+          : profileSharingStatus.sharedWithUrls,
+        sharedWithoutUrls: !shareNetworkUrls
+          ? true
+          : profileSharingStatus.sharedWithoutUrls,
+      };
```

Comment:
I find this difficult to decipher with the ternary; maybe this is simpler:
```javascript
const newProfileSharingStatus = {
  sharedWithUrls: shareNetworkUrls || profileSharingStatus.sharedWithUrls,
  sharedWithoutUrls: !shareNetworkUrls || profileSharingStatus.sharedWithoutUrls,
};
```
Also wondering if `urlShareOption` should be used instead of `shareNetworkUrls`. But I can't really think with all the boolean states happening...